src/rust_api.rs

```diff
@@ -284,7 +288,7 @@ mod tests {
         };
     }
 
-    fn dna(dna: &str) -> DnaSequence {
+    fn dna(dna: &str) -> DnaSequence<NucleotideAmbiguous> {
         DnaSequence::from_str(dna).unwrap()
     }
```

Comment:
I hate to say it since it'll be a pain, but I think the tests that use `dna` should be migrated to test on both `Nucleotide` and `NucleotideAmbiguous` to ensure the behavior is consistent.

Reply:
Also not sure if you've used quickcheck, but if you have or are interested in trying it, I think it could be useful to have a property test that tests behavior is consistent in the general case for `DnaSequence<Nucleotide>` and `DnaSequence<NucleotideAmbiguous>` where each `NucleotideAmbiguous` is A, T, C, or G.

If you want to write it and haven't used quickcheck before, LMK and I can point you at our property tests in the private repos and give some pointers.